File: dailystreak/LC-396(1stMay2026)/optimal.py


class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        # Time Complexity: O(n)
        # Space Complexity: O(1)
        
        n = len(nums)
        s = 0
        for i in nums:
            s += i

        b = 0
        for i in range(n):
            b += i * nums[i]
        maxx = b

        for i in range(n - 1, 0, -1):
            b = b + s - (n * nums[i])

            if b > maxx:
                maxx = b

        return maxx

File: dailystreak/LC-3300(29thMay26)/optimal.java
// Time Complexity: O(n × d) where d is the number of digits in the largest number. Since d ≤ 5, it simplifies to O(n).

// Space Complexity: O(1) because no extra data structures are used.


class Solution {
    public int minElement(int[] nums) {
        int min = Integer.MAX_VALUE;

        for (int num : nums) {
            int sum = 0;

            while (num > 0) {
                sum += num % 10;
                num /= 10;
            }

            min = Math.min(min, sum);
        }

        return min;
    }
}